jkube-kit/doc/src/main/asciidoc/inc/enricher/pod-annotation/_jkube_pod_annotation.adoc

```diff
@@ -0,0 +1,5 @@
+[[jkube-pod-annotation]]
+==== jkube-pod-annotation
+
+This enricher copies the annotations from a Controller (Deployment/ReplicaSet/StatefulSet) metadata to the annotations of
```

Comment:
I think this is not true ATM (only Deployment and DeploymentConfig are considered). Also, maybe the platform checks aren't necessary.

Maybe we can improve the ControllerHandler to include a method `PodTemplateSpec getPodTemplate(T controller);` that we can reuse here.

Also maybe HandlerHub could have a centralized `ControllerHandler<T> getHandlerFor(T hasMetadata)` that can be used to retrieve the appropriate ControllerHandler.

Then it would be as easy as replacing the logic in the loop with:
```
final ControllerHandler<> controllerHandler = xxx.getHandlerFor(item);
if (controllerHandler != null) {
  final PodTemplateSpec template = controllerHandler.getPodTemplate(item);
  // do the annotation stuff
```

Can you do this? Shall I take care of it?

Reply:
I can take care of this. Just to be clear, are you talking about removing this enricher and moving the annotation-related logic to handlers?

Reply:
Nope, I'm talking about taking advantage of the handlers to retrieve the PodTemplateSpec
The long snippet belongs to the PodAnnotationEnricher.